IA/IA_upload.py

```diff
@@ -19,91 +23,66 @@
 HERE = os.path.dirname(os.path.abspath(__file__))
 
 
-def mp_from_ids(mp_id: str, mp_keyname: str, bucket: str) -> MultiPartUpload:
-    mp = MultiPartUpload(bucket)
-    mp.key_name = mp_keyname
-    mp.id = mp_id
-    return mp
-
-
-async def gather_and_upload(bucket_name: str, parent: str):
+def gather_and_upload(item_name: str, parent: str, guid: str):
     '''
-    This script traverses through a directory uploading everything in it to Internet Archive.
+    This script uploads everything in a directory to Internet Archive.
     '''
 
-    tasks = []
-
-    directory_path = os.path.join(HERE, parent)
-    for root, dirs, files in os.walk(directory_path):
-        for file in files:
-            path = os.path.join(root, file)
-            with open(path, 'rb') as fp:
-                data = fp.read()
-                size = len(data)
-                url_path = path.split(parent+'/')[1]
-                if size > CHUNK_SIZE:
-                    tasks.append(chunked_upload(bucket_name, url_path, data))
-                else:
-                    tasks.append(upload(bucket_name, url_path, data))
-
-    await asyncio.gather(*tasks)
-
-
-async def upload(bucket_name: str, filename: str, file_content: bytes):
-    headers = {
-        'authorization': 'LOW {}:{}'.format(IA_ACCESS_KEY, IA_SECRET_KEY),
-        'x-amz-auto-make-bucket': '1',
-        'Content-Type': 'application/octet-stream',
-        'x-archive-meta01-collection': OSF_COLLECTION_NAME,
-    }
-    url = f'{IA_URL}/{bucket_name}/{filename}'
-    resp = put_with_retry(f'http://{url}', headers=headers, data=file_content, retry_on=(429, 503))
+    session_data = {'s3': {'access': IA_ACCESS_KEY, 'secret': IA_SECRET_KEY}}
+    session = internetarchive.get_session(config=session_data)
 
-    if resp.status_code != 200:
-        error_json = dict(xmltodict.parse(resp.content))
-        raise requests.exceptions.HTTPError(error_json)
+    ia_item = session.get_item(item_name)
 
+    directory_path = os.path.join(HERE, parent)
 
-async def chunked_upload(bucket_name: str, filename: str, file_content: bytes):
-    conn = boto.connect_s3(
-        IA_ACCESS_KEY,
-        IA_SECRET_KEY,
-        host=IA_URL,
-        is_secure=False,
-        calling_format=OrdinaryCallingFormat(),
+    ia_upload(ia_item, directory_path + '/', session)
```

Comment:
This is nice to be able to offload the directory crawling to the IA function.

Reply:
Agreed! Wish this was better documented in the AI source code. Could you add a comment explaining this with a link to https://archive.org/services/docs/api/internetarchive/quickstart.html? (or wherever else you learned this works)